api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java

```diff
@@ -243,4 +241,132 @@ public Bed getLatestBedByVisit(String visitUuid) {
+    public AdmissionLocation getAdmissionLocationsByLocation(Location location) {
+        Session session = sessionFactory.getCurrentSession();
+        Set<Location> locations = new HashSet<Location>(Arrays.asList(location));
+        Set<Location> childLocations = location.getChildLocations();
+        if (!CollectionUtils.isEmpty(childLocations)) {
+            locations.addAll(childLocations);
+        }
+
+        String hql = "select count(blm.bed) as totalBeds ," +
+                " COALESCE(sum(CASE WHEN blm.bed IS NOT NULL AND blm.bed.status = :occupied THEN 1 ELSE 0 END), 0) as occupiedBeds" +
+                " from BedLocationMapping blm where blm.location in (:locations)";
+
+        AdmissionLocation admissionLocation = (AdmissionLocation) session.createQuery(hql)
+                .setParameterList("locations", locations)
+                .setParameter("occupied", BedStatus.OCCUPIED.toString())
+                .setResultTransformer(Transformers.aliasToBean(AdmissionLocation.class))
+                .uniqueResult();
+        List<BedLayout> bedLayouts = getBedLayoutByLocation(location);
+
+        admissionLocation.setWard(location);
+        admissionLocation.setBedLayouts(bedLayouts);
+        return admissionLocation;
+    }
+
+    @Override
+    public List<BedLocationMapping> getBedLocationMappingByLocation(Location location) {
+        String hql = "select blm " +
+                "from BedLocationMapping blm " +
+                "where blm.location=:location";
+
+        Query query = sessionFactory.getCurrentSession().createQuery(hql);
+        query.setParameter("location", location);
+        return query.list();
+    }
+
+    @Override
+    public BedLocationMapping getBedLocationMappingByLocationAndRowAndColumn(Location location, Integer row, Integer column) {
+        String hql = "select blm " +
+                "from BedLocationMapping blm " +
+                "where blm.location=:location " +
+                "and blm.row=:row and blm.column=:column";
+
+        Query query = sessionFactory.getCurrentSession().createQuery(hql);
+        query.setParameter("location", location);
+        query.setParameter("row", row);
+        query.setParameter("column", column);
+        return (BedLocationMapping) query.uniqueResult();
+    }
+
+    @Override
+    public BedLocationMapping saveBedLocationMapping(BedLocationMapping bedLocationMapping) {
+        Session session = this.sessionFactory.getCurrentSession();
+        session.saveOrUpdate(bedLocationMapping);
+        session.flush();
+        return bedLocationMapping;
+    }
+
+    @Override
+    public List<Integer> getAdmissionLocationIds() {
+        Session session = sessionFactory.getCurrentSession();
+        String sql = "SELECT ltm.location_id\n" +
+                "  FROM location_tag_map ltm\n" +
+                "    LEFT JOIN location_tag lt ON ltm.location_tag_id = lt.location_tag_id\n" +
+                "  WHERE lt.name = 'Admission Location'";
```

Comment:
@sanishmaharjan in the original code the tag `'Admission Location'` was maintained in a constants class. Why did you re-hardcode it here?